### ISSUE TITLE
When I use cupy.cuda.cudnn in Python2, it raises CUDNN_STATUS_EXECUTION_FAILED.

### ISSUE THREAD

**Issue by mqcmd196** [cupy/cupy#3358]
Hi! When I use cupy.cuda.cudnn in Python2, it raises the error and I couldn't debug it. I know Python2 support had finished, but I need to use some reasons. So please give me advice if you can. I'll show the log.

```bash
rosrun jsk_perception train_ssd.py --train-dataset-dir ./train/dataset_voc/ --val-dataset-dir ./test/dataset_voc/
/usr/lib/python2.7/dist-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
chainer_mask_rcnn cannot be imported.
Exception in main training loop: CUDNN_STATUS_EXECUTION_FAILED
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/chainer/training/trainer.py", line 316, in run
    update()
  File "/usr/local/lib/python2.7/dist-packages/chainer/training/updaters/standard_updater.py", line 175, in update
    self.update_core()
  File "/usr/local/lib/python2.7/dist-packages/chainer/training/updaters/standard_updater.py", line 187, in update_core
    optimizer.update(loss_func, *in_arrays)
  File "/usr/local/lib/python2.7/dist-packages/chainer/optimizer.py", line 864, in update
    loss = lossfun(*args, **kwds)
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/home/yoshiki/research_ws/src/jsk_recognition/jsk_perception/scripts/train_ssd.py", line 69, in forward
    mb_locs, mb_confs = self.model(imgs)
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/chainercv/links/model/ssd/ssd.py", line 130, in forward
    return self.multibox(self.extractor(x))
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/chainercv/links/model/ssd/ssd_vgg16.py", line 203, in forward
    ys = super(VGG16Extractor512, self).forward(x)
  File "/usr/local/lib/python2.7/dist-packages/chainercv/links/model/ssd/ssd_vgg16.py", line 70, in forward
    h = F.relu(self.conv1_1(x))
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/chainer/links/connection/convolution_2d.py", line 184, in forward
    groups=self.groups)
  File "/usr/local/lib/python2.7/dist-packages/chainer/functions/connection/convolution_2d.py", line 589, in convolution_2d
    y, = fnode.apply(args)
  File "/usr/local/lib/python2.7/dist-packages/chainer/function_node.py", line 321, in apply
    outputs = self.forward(in_data)
  File "/usr/local/lib/python2.7/dist-packages/chainer/function_node.py", line 512, in forward
    return self.forward_gpu(inputs)
  File "/usr/local/lib/python2.7/dist-packages/chainer/functions/connection/convolution_2d.py", line 189, in forward_gpu
    return self._forward_cudnn(x, W, b, y)
  File "/usr/local/lib/python2.7/dist-packages/chainer/functions/connection/convolution_2d.py", line 250, in _forward_cudnn
    auto_tune=auto_tune, tensor_core=tensor_core)
  File "cupy/cudnn.pyx", line 1575, in cupy.cudnn.convolution_forward
  File "cupy/cuda/cudnn.pyx", line 1211, in cupy.cuda.cudnn.convolutionForward
  File "cupy/cuda/cudnn.pyx", line 715, in cupy.cuda.cudnn.check_status
Will finalize trainer extensions and updater before reraising the exception.
Traceback (most recent call last):
  File "/home/yoshiki/research_ws/src/jsk_recognition/jsk_perception/scripts/train_ssd.py", line 252, in <module>
    main()
  File "/home/yoshiki/research_ws/src/jsk_recognition/jsk_perception/scripts/train_ssd.py", line 248, in main
    trainer.run()
  File "/usr/local/lib/python2.7/dist-packages/chainer/training/trainer.py", line 349, in run
    six.reraise(*exc_info)
  File "/usr/local/lib/python2.7/dist-packages/chainer/training/trainer.py", line 316, in run
    update()
  File "/usr/local/lib/python2.7/dist-packages/chainer/training/updaters/standard_updater.py", line 175, in update
    self.update_core()
  File "/usr/local/lib/python2.7/dist-packages/chainer/training/updaters/standard_updater.py", line 187, in update_core
    optimizer.update(loss_func, *in_arrays)
  File "/usr/local/lib/python2.7/dist-packages/chainer/optimizer.py", line 864, in update
    loss = lossfun(*args, **kwds)
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/home/yoshiki/research_ws/src/jsk_recognition/jsk_perception/scripts/train_ssd.py", line 69, in forward
    mb_locs, mb_confs = self.model(imgs)
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/chainercv/links/model/ssd/ssd.py", line 130, in forward
    return self.multibox(self.extractor(x))
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/chainercv/links/model/ssd/ssd_vgg16.py", line 203, in forward
    ys = super(VGG16Extractor512, self).forward(x)
  File "/usr/local/lib/python2.7/dist-packages/chainercv/links/model/ssd/ssd_vgg16.py", line 70, in forward
    h = F.relu(self.conv1_1(x))
  File "/usr/local/lib/python2.7/dist-packages/chainer/link.py", line 294, in __call__
    out = forward(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/chainer/links/connection/convolution_2d.py", line 184, in forward
    groups=self.groups)
  File "/usr/local/lib/python2.7/dist-packages/chainer/functions/connection/convolution_2d.py", line 589, in convolution_2d
    y, = fnode.apply(args)
  File "/usr/local/lib/python2.7/dist-packages/chainer/function_node.py", line 321, in apply
    outputs = self.forward(in_data)
  File "/usr/local/lib/python2.7/dist-packages/chainer/function_node.py", line 512, in forward
    return self.forward_gpu(inputs)
  File "/usr/local/lib/python2.7/dist-packages/chainer/functions/connection/convolution_2d.py", line 189, in forward_gpu
    return self._forward_cudnn(x, W, b, y)
  File "/usr/local/lib/python2.7/dist-packages/chainer/functions/connection/convolution_2d.py", line 250, in _forward_cudnn
    auto_tune=auto_tune, tensor_core=tensor_core)
  File "cupy/cudnn.pyx", line 1575, in cupy.cudnn.convolution_forward
  File "cupy/cuda/cudnn.pyx", line 1211, in cupy.cuda.cudnn.convolutionForward
  File "cupy/cuda/cudnn.pyx", line 715, in cupy.cuda.cudnn.check_status
cupy.cuda.cudnn.CuDNNError: CUDNN_STATUS_EXECUTION_FAILED
```

I also exported debug.log.
[cudnn_dubug.log](https://github.com/cupy/cupy/files/4665657/cudnn_dubug.log)

These are my conditions. I installed cupy-cuda91 by pip2.
`python -c 'import cupy; cupy.show_config()'` shows
```
CuPy Version          : 6.7.0
CUDA Root             : /usr
CUDA Build Version    : 9010
CUDA Driver Version   : 10020
CUDA Runtime Version  : 9010
cuDNN Build Version   : 7102
cuDNN Version         : 7102
NCCL Build Version    : 2115
NCCL Runtime Version  : (unknown)
```

`nvcc -V` shows
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Nov__3_21:07:56_CDT_2017
Cuda compilation tools, release 9.1, V9.1.85
```

OS:Ubuntu 18.04.4 64bit

Thanks.

**Comment by kmaehashi** [comment]
This seems like an issue in Chainer. Could you raise an issue in https://github.com/chainer/chainer ? Please attach the code If possible.

Also I don't recommend using CUDA 9.1 as the latest cuDNN (which may include bug fix for this issue) is not supported.

**Comment by 821736960** [comment]
hi, did you slove the problem? i met the same problem as you.

**Comment by pazeshun** [comment]
@mqcmd196 How did you solve this error?

I faced the same error with the similar environment (Ubuntu 18.04.6, cuda (9.1) from nvidia-cuda-toolkit apt package, Python2.7.17, cupy-cuda91==6.7.0, chainer==6.7.0).
When I tried installing cupy instead of cupy-cuda91 to exclude cuDNN, I faced a different error (sorry but I forgot it).
Finally those errors are disappeared when I installed cuda==10.2, cuDNN==v7.6.5, cupy==6.7.0, chainer==6.7.0.
But this is not a good status because cupy==6.7.0 does not officially support cuda==10.2 and cuDNN==v7.6.5:
https://docs.cupy.dev/en/v6.7.0/install.html#requirements
https://docs.cupy.dev/en/v6.7.0/install.html#optional-libraries

Why I tried upgrading cuda is that I found someone says upgrading cuda solved the issue while I was googling (sorry again but I forgot the source URL).
Installing cuda==9.2 by https://developer.nvidia.com/cuda-92-download-archive?target_os=Linux&target_arch=x86_64&target_distro=Ubuntu&target_version=1710&target_type=deblocal failed because this procedure uninstalled `ubuntu-desktop` and many other packages.
Installing cuda==10.1 succeeded, but `import cupy` with cupy-cuda101==6.7.0 raised `ImportError: CuPy is not correctly installed`.
`sudo pip install cupy==6.7.0 --no-cache-dir` also failed due to some header files (e.g., `cublas_v2.h`) existed in cuda 10.2 folder (`/usr/local/cuda-10.2/targets/x86_64-linux/include`).
Finally, I tried cuda==10.2 and succeeded.
Just for your information, cuDNN>=v8.0.0 was ignored by cupy installation.
cuDNN==v7.6.5 is the only version supporting cuda==10.2 under v8.0.0.

**Comment by kmaehashi** [comment]
For those looking for working Chainer environment with CUDA & cuDNN, I'd suggest using docker images which is available for each Chainer version & Python 2.7/3.x:

https://hub.docker.com/r/chainer/chainer/tags
